seacrowd/sea_datasets/thai_nner/thai_nner.py

```diff
@@ -0,0 +1,594 @@
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_seacrowd_seq_label",
+            version=SEACROWD_VERSION,
+            description=f"{_DATASETNAME} SEACrowd schema",
+            schema="seacrowd_seq_label",
+            subset_id=f"{_DATASETNAME}",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "thai_nner_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features({"index": datasets.Value("string"), "tokens": [datasets.Value("string")], "ner_tag": [datasets.Value("string")]})
+
+        elif self.config.schema == "seacrowd_seq_label":
+            features = schemas.seq_label_features(self.label_classes)
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        train_path = Path(dl_manager.download_and_extract(_URLS["train"]))
+        test_path = Path(dl_manager.download_and_extract(_URLS["test"]))
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": train_path,
+                    "split": "train",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={
+                    "filepath": test_path,
+                    "split": "test",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath: Path, split: str) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+
+        def _custom_conll_loader(file_path):
+            # Read file
+            data = open(file_path, "r", encoding="utf-8").readlines()
+
+            # Prepare buffer
+            dataset = []
+            sentence, seq_label = [], []
+            for line in data:
+                if len(line.strip()) > 0:
+                    # Handle lines with a single field
+                    if "\t" in line:
+                        token, label = line[:-1].split("\t")
+                    elif "\t" not in line and " " in line:
+                        line_split = line.split()
+                        token = line_split[0]
+                        label = line_split[1:]
+                    else:
+                        token, label = line.strip(), "O"  # Default label if not provided
+
+                    sentence.append(token)
+                    seq_label.append(label)
+                else:
+                    dataset.append({"sentence": sentence, "label": seq_label})
+                    sentence = []
+                    seq_label = []
+            return dataset
```

Comment:
Hi @bp-high! Thanks for the contribution!

I only want to address a minor bug that might happen here since the implementation hasn't passed the test case.

```
THIS IS YOUR `row['sentence']`
['5', '_', 'มี', '.ค.', '_']

THIS IS YOUR `row['label']`
[['B-date', 'S-day', 'O', 'O', 'O', 'O', 'O', 'O'], ['I-date', 'O', 'O', 'O', 'O', 'O', 'O', 'O'], ['I-date', 'B-month', 'O', 'O', 'O', 'O', 'O', 'O'], ['I-date', 'E-month', 'O', 'O', 'O', 'O', 'O', 'O'], ['I-date', 'O', 'O', 'O', 'O', 'O', 'O', 'O']]
```

I think there is a slight mistake within the loader function resulting in the captured `label` becomes list of list instead of list of string, sequentially matched with each token from the sentence. Hence, it prompts `TypeError: '<=' not supported between instances of 'int' and 'list'` during the check.

Could you please do revision on this? Other than that, LGTM!

Reply:
Nope this is expected as it is a depth based NER(NNER/ nested NER) with each word having eight depth based ner tags. So the list belongs to each word.
![Screenshot 2024-04-26 at 4 08 59 AM](https://github.com/SEACrowd/seacrowd-datahub/assets/53102161/9b2ff509-adf0-4707-837f-16045e2e8457)

Reply:
The loader is specially modified to give in the format so that we get all labels(upto depth 8) of each word in a corresponding list marked for it.